### PR TITLE
Reformat YAML data

### PR DESCRIPTION
**Pull request by Jacob-Allebach**
# Reformat YAML data

## Description

This pull request will add the code that reformats the raw data taken from the YAML file into something much easier to work with.

## Type of Change

- [x] Feature
- [ ] Bug fix
- [ ] Documentation

## Contributors

- @Jacob-Allebach 

## Reminder

 - All GitHub Actions should be in a passing state before any pull request is merged.
 - All PRs must be reviewed by at least one team member and one member of the Integration team!
 - Any issues this PR closes are tagged in the description!

### CODE REVIEW

**Comment by mariakimheinert** [comment]
Hello @Jacob-Allebach, since this is a source code contribution, please remember to add a member from the Integration team as a reviewer as well. Additionally, please also let me know when you would like an industry professional to be added as a reviewer. Thanks!

File: gatorgrade/input/in_file_path.py
Comment (noorbuchi) [inline gatorgrade/input/in_file_path.py:16]: If `none` is being passed to the function, is that argument really needed? You might have other uses for it, It's just not clear just by looking at the code. Also, it's generally not a good habit to pass `None` into functions, so if it can be avoided, let's try that.

File: gatorgrade/input/in_file_path.py
Comment (noorbuchi) [inline gatorgrade/input/in_file_path.py:17]: Since `loop_through_data` doesn't return anything, it doesn't have an output and so you're essentially returning the same `reformatted_data` before making the call to that function.

File: gatorgrade/input/in_file_path.py
Comment (noorbuchi) [inline gatorgrade/input/in_file_path.py:20]: I think a more descriptive name of this function is needed to better describe what it does.

File: gatorgrade/input/in_file_path.py
Comment (noorbuchi) [inline gatorgrade/input/in_file_path.py:34]: This function currently doesn't return anything, should something be returned here?

File: gatorgrade/input/in_file_path.py
Comment (noorbuchi) [inline gatorgrade/input/in_file_path.py:25]: there is a better way to check for this condition. You can use the `isinstance` function to write something like this:

```python
if isinstance(dict[item], list):

```

File: gatorgrade/input/in_file_path.py
Comment (noorbuchi) [inline gatorgrade/input/in_file_path.py:26]: This can be rephrased to say:

```python
if not path:
```

**Review by noorbuchi — CHANGES_REQUESTED** [review]
Thanks for all your work! Here are some questions/suggestions for you as you continue to work on this pull request.

**Comment by Jacob-Allebach** [comment]
@noorbuchi I've edited a bit of the code based on your suggestions, but I just wanted to clarify a couple of your other points. I believe the `loop_through_data` function (now renamed `add_checks_to_list`) does have to take in `None` as an input because in the case that there is no file path, it has to put `None` as the `"file_context"` value in the resulting dictionary. There might be a way around that, but I don't currently see it. Also, the function doesn't return anything because it's adding the data for the checks to `reformatted_data` in each recursive loop. Is there a clearer way of documenting that in the code?

File: gatorgrade/input/in_file_path.py
Comment (laurennevill) [inline gatorgrade/input/in_file_path.py:34]: This code all looks pretty good! The only think I'm noticing is an overall lack of source code comments. This function specifically has none. Source code comments are a great method of documentation and can make the code easier to understand especially for those who might find themselves revisiting this in the future. If you could take some time to write some useful comments I think it would greatly improve the sustainability of this work. Otherwise, good work!

**Review by laurennevill — CHANGES_REQUESTED** [review]

File: gatorgrade/input/in_file_path.py
Comment (Jacob-Allebach) [reply gatorgrade/input/in_file_path.py]: I just added some comments to this function that will hopefully make it better documented. Let me know if you think anything else could be added to improve the documentation in the code.

File: gatorgrade/input/in_file_path.py
Comment (tuduun) [inline gatorgrade/input/in_file_path.py:16]: Hey Jacob, your code looks great so far. There were no linting errors. However, are we not going to need the shell setup commands? Or are you going to use the setup shell commands data and pass it in your next commit or next PR?

Comment (Jacob-Allebach) [reply gatorgrade/input/in_file_path.py:16]: The shell commands will be acquired before this function is run, so in between running the `parse_yaml_file` function and the `reformat_yaml_data` function, we can get the data from the output of `parse_yaml_file` by checking if there are 2 values in the list, and if there are, the first value will be the setup commands. Those commands are stored as a dictionary with the key being `"setup"` and the value being a singular string of each setup command. The different commands for setup are separated in the string by `\n`, which represents a new line. Let me know if you need any more info about that!

File: gatorgrade/input/in_file_path.py
Comment (laurennevill) [reply gatorgrade/input/in_file_path.py]: The added comments look great, thank you for taking the time to work them in.

**Review by Michionlion — CHANGES_REQUESTED** [review]
Great work, @Jacob-Allebach! I've got two minor suggestions and a request for you to refactor `add_checks_to_list` slightly to rely on pass-by-reference for recursive data management. None of them are critical, so if you deem the value in doing any of them to be not worth it, say so (and describe why so that future reviewers can understand) and re-request review.

**Comment by mariakimheinert** [comment]
Hello @Jacob-Allebach! Since there are now a few other PRs, such as #58, #64, and #65, that are waiting for this PR to be merged, could you please give an update on the status of this PR? Please make sure to resolve conversations that you have already resolved and if there are any non-critical requested changes that you believe would be better suited for future work (perhaps @Michionlion's feedback), you should document these requested changes in an issue and resolve the conversations so that you can move towards merging this PR.

**Comment by Jacob-Allebach** [comment]
> Hello @Jacob-Allebach! Since there are now a few other PRs, such as #58, #64, and #65, that are waiting for this PR to be merged, could you please give an update on the status of this PR? Please make sure to resolve conversations that you have already resolved and if there are any non-critical requested changes that you believe would be better suited for future work (perhaps @Michionlion's feedback), you should document these requested changes in an issue and resolve the conversations so that you can move towards merging this PR.

I just created an issue with all of the requested changes to be worked on in the future so that this pull request can get merged and others can begin merging theirs. Should I just resolve the conversations since they are now in the issue?

**Review by Michionlion — APPROVED** [review]
Given that there are no critical issues with this PR, I think it is ready to merge! Thank you for documenting those minor changes, @Jacob-Allebach, and great work!

**Review by laurennevill — APPROVED** [review]
good work!

**Review by mariakimheinert — APPROVED** [review]
@Jacob-Allebach these changes look great. Thanks for all your hard work on making this contribution to GatorGrade! Once one of your team members also approves this PR, it is ready to be merged :shipit:

**Review by tuduun — APPROVED** [review]

**Pull request merged**
